File: Easy/268-missing-number/try-3.cpp
/*
 * Problem #268: Missing Number
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 18/01/2025, 15:23:01
 * Link: https://leetcode.com/problems/missing-number/
 */

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int xor1 = 0,i;
        for(i = 0;i<nums.size();i++){
            xor1 = xor1^nums[i];
            xor1 = xor1^(i+1);
        }
        return xor1;
    }
};

File: Easy/2239-find-closest-number-to-zero/try-3.py
/*
 * Problem #2239: Find Closest Number to Zero
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 13/12/2024, 05:51:43
 * Link: https://leetcode.com/problems/find-closest-number-to-zero/
 */

class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        mx=float('inf')
        for i in nums:
            if (abs(mx)>abs(i)):
                mx=i
            elif((abs(mx)==abs(i))):
                mx=max(mx,i)
 
        return mx
            

